src/vs/workbench/contrib/terminal/browser/terminalStatusList.ts

```diff
@@ -70,7 +70,9 @@ export class TerminalStatusList extends Disposable implements ITerminalStatusLis
 		let result: ITerminalStatus | undefined;
 		for (const s of this._statuses.values()) {
 			if (!result || s.severity >= result.severity) {
-				result = s;
+				if (s.icon || !result?.icon) {
```

Comment:
though i'm sure this could be combined with the above condition, I think that would make it more confusing.